homeassistant/components/litterrobot/vacuum.py

```diff
@@ -85,19 +99,10 @@ class LitterRobotCleaner(LitterRobotControlEntity, StateVacuumEntity):
     @property
     def state(self) -> str:
         """Return the state of the cleaner."""
-        switcher = {
-            LitterBoxStatus.CLEAN_CYCLE: STATE_CLEANING,
-            LitterBoxStatus.EMPTY_CYCLE: STATE_CLEANING,
-            LitterBoxStatus.CLEAN_CYCLE_COMPLETE: STATE_DOCKED,
-            LitterBoxStatus.CAT_SENSOR_TIMING: STATE_DOCKED,
-            LitterBoxStatus.DRAWER_FULL_1: STATE_DOCKED,
-            LitterBoxStatus.DRAWER_FULL_2: STATE_DOCKED,
-            LitterBoxStatus.READY: STATE_DOCKED,
-            LitterBoxStatus.CAT_SENSOR_INTERRUPTED: STATE_PAUSED,
-            LitterBoxStatus.OFF: STATE_OFF,
-        }
+        if self.robot.last_seen < datetime.now(timezone.utc) - UNAVAILABLE_AFTER:
+            return STATE_UNAVAILABLE
```

Comment:
Please implement the `available` property instead. We don't use `STATE_UNAVAILABLE` directly.

Reply:
I missed it in the review

Reply:
Would it be better to use STATE_UNKNOWN instead? The only think I don't like about using the `available` property is that it hides all of the `extra_state_attributes` when it is unavailable?

Reply:
Those two states have different meanings. Only one can be correct.

Reply:
> Would it be better to use STATE_UNKNOWN instead? The only think I don't like about using the `available` property is that it hides all of the `extra_state_attributes` when it is unavailable?

Regardless of the unknown state, it is recommended not to use `extra_state_attributes`, looking at the attributes set there some of them are frequently changing, especially the `last_seen`, every time one of them is changed a new state is generated and recorded. It is better to move them to sensors categorized as diagnostics (so they are not added to dashboard by default) or even set some of them to be disabled by default.

Reply:
Thanks, that was gonna be my next question. I'll move that into a separate sensor.